### PR TITLE
Add custom dispatcher to builtin-json.

### PR DESCRIPTION
**Pull request by orkvi**
JerryScript-DCO-1.0-Signed-off-by: Orkenyi Virag orkvi@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-json.c [hunk 1]
```diff
@@ -831,11 +846,9 @@ ecma_builtin_json_parse_buffer (const lit_utf8_byte_t * str_start_p, /**< String
  *         Returned value must be freed with ecma_free_value.
  */
 static ecma_value_t
-ecma_builtin_json_parse (ecma_value_t this_arg, /**< 'this' argument */
-                         ecma_value_t arg1, /**< string argument */
+ecma_builtin_json_parse (ecma_value_t arg1, /**< string argument */
                          ecma_value_t arg2) /**< reviver argument */
 {
-  JERRY_UNUSED (this_arg);
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-json.c:838]: Remove newline after this. You did it right in the seocnd case.

File: jerry-core/ecma/builtin-objects/ecma-builtin-json.c [hunk 2]
```diff
@@ -1789,6 +1799,41 @@ ecma_builtin_json_stringify (ecma_value_t this_arg, /**< 'this' argument */
...
+ecma_builtin_json_dispatch_routine (uint8_t builtin_routine_id, /**< built-in wide routine identifier */
+                                     ecma_value_t this_arg, /**< 'this' argument value */
+                                     const ecma_value_t arguments_list_p[], /**< list of arguments
+                                                                             *   passed to routine */
+                                     uint32_t arguments_number) /**< length of arguments' list */
+{
+  JERRY_UNUSED (this_arg);
+  ecma_value_t arg0 = (arguments_number > 0)? arguments_list_p[0]:ECMA_VALUE_UNDEFINED;
+  ecma_value_t arg1 = (arguments_number > 1)? arguments_list_p[1]:ECMA_VALUE_UNDEFINED;
+  ecma_value_t arg2 = (arguments_number > 2)? arguments_list_p[2]:ECMA_VALUE_UNDEFINED;
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-json.c:1818]: Syntax is incorrect, put spaces around ? and :

**Review by zherczeg — CHANGES_REQUESTED** [review]
Only minor things.

**Review by zherczeg — APPROVED** [review]
LGTM

File: jerry-core/ecma/builtin-objects/ecma-builtin-json.c [hunk 3]
```diff
@@ -1789,6 +1798,41 @@ ecma_builtin_json_stringify (ecma_value_t this_arg, /**< 'this' argument */
...
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+ecma_value_t
+ecma_builtin_json_dispatch_routine (uint8_t builtin_routine_id, /**< built-in wide routine identifier */
+                                     ecma_value_t this_arg, /**< 'this' argument value */
+                                     const ecma_value_t arguments_list_p[], /**< list of arguments
+                                                                             *   passed to routine */
+                                     uint32_t arguments_number) /**< length of arguments' list */
```
Comment (galpeter) [inline jerry-core/ecma/builtin-objects/ecma-builtin-json.c:1812]: Incorrect indentation

File: jerry-core/ecma/builtin-objects/ecma-builtin-json.c [hunk 4]
```diff
@@ -1789,6 +1798,41 @@ ecma_builtin_json_stringify (ecma_value_t this_arg, /**< 'this' argument */
...
+{
+  JERRY_UNUSED (this_arg);
+  ecma_value_t arg0 = (arguments_number > 0) ? arguments_list_p[0] : ECMA_VALUE_UNDEFINED;
+  ecma_value_t arg1 = (arguments_number > 1) ? arguments_list_p[1] : ECMA_VALUE_UNDEFINED;
+  ecma_value_t arg2 = (arguments_number > 2) ? arguments_list_p[2] : ECMA_VALUE_UNDEFINED;
```
Comment (galpeter) [inline jerry-core/ecma/builtin-objects/ecma-builtin-json.c:1817]: To keep it in sync with the methods (parse, stringify) we should start the naming from 1 (arg0 should be arg1, etc..)

**Review by galpeter — CHANGES_REQUESTED** [review]

**Review by galpeter — APPROVED** [review]
lgtm

**Pull request merged**